infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrack.kt

```diff
@@ -54,17 +83,158 @@ enum class LocationTrackState(val category: LayoutStateCategory) {
     fun isRemoved() = this == DELETED
 }
 
+private const val MISSING_PART_PLACEHOLDER = "???"
+
+private fun getShortNumber(switch: LayoutSwitch?): String =
+    switch?.nameParts?.shortNumberPart?.toString() ?: MISSING_PART_PLACEHOLDER
+
+private fun getShortName(switch: LayoutSwitch?): String =
+    switch?.nameParts?.let { "${it.prefix} ${it.shortNumberPart}" } ?: MISSING_PART_PLACEHOLDER
```

Comment:
En muista suorilta onko näille käyttötarvetta missään muualla, mutta kun kyseessä on vaihteen nimen mälväys, niin nämä funktiot voisivat mielestäni löytyä suoraan `LayoutSwitch`:iltä. Sillähän oli aiemmin tällä reviewillä poistunut `shortName`-property jonka poistaminen oli kyllä hyvä idea, mutta veikkaisin että näille saattaisi olla tulevaisuudessa tarvetta muuallakin kuin sijaintiraiteen sisällä

Reply:
Hmm, nojoo, kyllähän nuo voisi sinne siirtää, lukuun ottamatta tuota placeholderia. Se on minusta aivan selvästi raiteen nimen/kuvauksen tekstiesityksen juttua, ei mitään yleistä, mutta voisihan nämä muutoin olla vaihteella lazy val:na. Siirränpä sinne.

Reply:
... ääh, enpäs ole oikein vakuuttunut. Nyt kun asiaa katsoo pidemmällä tuolla koodissa niin ehkä ne voisi olla tuon parsitun nimi-osa-olion kenttiä... mutta tuo numberPart on jo. Nimelle voisi sinne ehkä tehdä val:n joka on vain ne 2 osaa peräkäin kuten tässäkin -> seems silly. Ehkäpä en nyt tässä kohtaa muutakaan tuota minnekään. Katsotaan mielummin jossain kohtaa se tiksu jossa mietitään muutenkin että miten tuon logiikan saisi yhteen + lokalisointiin mukaan backarin ja frontin osalta.